pandas/conftest.py

```diff
@@ -409,16 +421,16 @@ def tz_aware_fixture(request):
 COMPLEX_DTYPES = [complex, "complex64", "complex128"]
 STRING_DTYPES = [str, 'str', 'U']
 
-DATETIME_DTYPES = ['datetime64[ns]', 'M8[ns]']
-TIMEDELTA_DTYPES = ['timedelta64[ns]', 'm8[ns]']
+DATETIME64_DTYPES = ['datetime64[ns]', 'M8[ns]']
+TIMEDELTA64_DTYPES = ['timedelta64[ns]', 'm8[ns]']
```

Comment:
Rationale for renames?

Is this necessary given how massive the diff is already?

Reply:
I can split off the conftest-related things, of course.

Regarding your specific point: The rename is really important IMO, because the `DATETIME_DTYPES` (as-is) have nothing to do with `datetime.dateime` or other such things.

However, when reading a test that is being fed by a `datetime_dtype` fixture, one could easily think that this tests `datetime.datetime`.

The current state in conftest specifically contains only `DATETIME64_DTYPES`, and I believe this distinction is used in many parts of the code base as well.

Reply:
Got it.  I'm thinking that we should separate this out.  The more we can condense this PR, the better.

But hold off for the moment (just one more question re: some of other things you've changed for `conftest.py`.

Reply:
I can change the `DATETIME_DTYPES` -> `DATETIME64_DTYPES` in a precursor of course, but the actual change that precipitates the necessity is the introduction of `datetime64_dtype`, which wouldn't be used before this PR. Would you want me to introduce the fixtures with the conftest-precursor, or in this PR?